### PR TITLE
finish making Expression.op const

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/opover.d [hunk 1]
```diff
@@ -478,9 +478,17 @@ private Expression checkAliasThisForRhs(AggregateDeclaration ad, Scope* sc, BinE
...
+ *      sc = context
+ *      pop = if not null, is set to the operator that was actually overloaded,
+ *            which may not be `e.op`. Happens when operands are reversed to
+ *            to match an overload
+ * Returns:
+ *      `null` if not an operator overload,
+ *      otherwise the lowered expression
  */
-Expression op_overload(Expression e, Scope* sc)
+Expression op_overload(Expression e, Scope* sc, TOK* pop = null)
```
Comment (JinShil) [inline src/dmd/opover.d]: You're violating some of your own principles here:  prefer `ref` to pointers, avoid default parameters.

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/opover.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/opover.d]: I know. I did it to minimize the diffs. Also, the new rvalue ref change will support defaulting ref parameters, which will nicely avoid this little bit of ugliness.

**Comment by jacob-carlborg** [comment]
This causes DMD to fail to build using LDC as the host compiler:

```
src/dmd/opover.d(1051): Error: function `dmd.opover.op_overload.OpOverload.visit` cannot access frame of function `dmd.opover.op_overload`
src/dmd/opover.d(1052): Error: function `dmd.opover.op_overload.OpOverload.visit` cannot access frame of function `dmd.opover.op_overload`
src/dmd/opover.d(1356): Error: function `dmd.opover.op_overload.OpOverload.visit` cannot access frame of function `dmd.opover.op_overload`
```

**Comment by WalterBright** [comment]
> This causes DMD to fail to build using LDC as the host compiler:

https://github.com/dlang/dmd/pull/9700

**Comment by jacob-carlborg** [comment]
@thewilsonator Semaphore **did** catch this regression: https://semaphoreci.com/dlang/dmd-2/branches/master/builds/1997, but it was merged anyway.

**Comment by thewilsonator** [comment]
Ah, that is my bad then. It shows the poor SNR the semaphore has w.r.t failures. I must have thought it was another non reproducible build problem.

**Pull request merged**